src/lib/LedgerApi.ts

```diff
@@ -326,39 +323,57 @@ class LedgerApi {
         }
 
         // prepare tx outside of request to avoid that an error would result in an endless loop in _callLedger
-        const senderPubKeyBytes = await LedgerApi.getPublicKey(keyPath);
+        const [senderPubKeyBytes] = await Promise.all([
+            LedgerApi.getPublicKey(keyPath, walletId),
+            Nimiq.WasmHelper.doImportBrowser() // wasm needed for toAddress
+                .catch((e) => LedgerApi._throwError(LedgerApi.ErrorType.FAILED_LOADING_DEPENDENCIES, e)),
+        ]);
         const senderPubKey = new Nimiq.PublicKey(senderPubKeyBytes);
-        const senderAddress = senderPubKey.toAddress().toUserFriendlyAddress();
-        if (transaction.sender.replace(/ /g, '')
-            !== senderAddress.replace(/ /g, '')) {
+        const senderAddress = senderPubKey.toAddress();
+        if (!senderAddress.equals(transaction.sender)) {
             LedgerApi._throwError(LedgerApi.ErrorType.REQUEST_ASSERTION_FAILED,
-                'Sender Address doesn\'t match this ledger account', request);
+                'Sender Address doesn\'t match this ledger address', request);
         }
+        const senderType = transaction.senderType !== undefined
+            ? transaction.senderType
+            : Nimiq.Account.Type.BASIC;
+
+        const recipientType = transaction.recipientType !== undefined
+            ? transaction.recipientType
+            : Nimiq.Account.Type.BASIC;
+
         const genesisConfig = Nimiq.GenesisConfig.CONFIGS[transaction.network];
         const networkId = genesisConfig.NETWORK_ID;
-        const recipient = Nimiq.Address.fromUserFriendlyAddress(transaction.recipient);
-        const value = Nimiq.Policy.coinsToSatoshis(transaction.value);
-        const fee = Nimiq.Policy.coinsToSatoshis(transaction.fee);
-        if (transaction.extraData && transaction.extraData.length !== 0) {
-            nimiqTx = new Nimiq.ExtendedTransaction(senderPubKey.toAddress(), Nimiq.Account.Type.BASIC,
-                recipient, Nimiq.Account.Type.BASIC, value - fee, fee, transaction.validityStartHeight,
-                Nimiq.Transaction.Flag.NONE, transaction.extraData, undefined, networkId);
+        const flags = transaction.flags !== undefined
+            ? transaction.flags
+            : Nimiq.Transaction.Flag.NONE;
+        const fee = transaction.fee || 0;
+
+        if (transaction.extraData && transaction.extraData.length !== 0
+            || senderType !== Nimiq.Account.Type.BASIC
+            || recipientType !== Nimiq.Account.Type.BASIC
+            || flags !== Nimiq.Transaction.Flag.NONE
+        ) {
+            const extraData = transaction.extraData ? transaction.extraData : new Uint8Array(0);
+            nimiqTx = new Nimiq.ExtendedTransaction(senderAddress, senderType, transaction.recipient, recipientType,
+                transaction.value - fee, fee, transaction.validityStartHeight, flags, extraData, undefined,
+                networkId);
```

Comment:
`networkId` can be left out, it defaults to the currently initialized GenesisConfig anyway.

Reply:
The `LegderApi` is meant to be usable by other devs outside of the accounts manager some day.
Therefore the support for specifying the network.

Reply:
Haha, but above you are setting it to fixed `networkId = genesisConfig.NETWORK_ID`, the same as leaving it out here would do. It's not even configurable.

Reply:
Then I'd say you need to make sure, as commented above, that `networkId` actually falls back to the configured GenesisConfig, instead of throwing an exception.